### ISSUE TITLE
the graph reasoning module issue

### ISSUE THREAD

**Issue by runzeer** [yuweijiang/HGL-pytorch#4]
Thanks for your sharing.Great work! But when I read the codes according to the papers, there exists some issues that I can not understand, especially the graph reasoning module.  In the henG.py file , I  can not find the impletation of the formula (6) in your paper.  
My first question:
The e_obj variable means the x_middle in your paper,right? But it is not the combination of Y_o and X_m with softmax.Perhaps it should be like this:
e_obj = self.fc_o_(torch.cat([s_obj, o_a_view], -1))    #49th line

My second question:
In formula (6), Y_v is the combination of X_middle and Y_o. But in your impletation,  it seems like the 
Y_v is the combination of X_middle and answer_view. So in my opinion,it should be like this:
A_obj = F.softmax(self.w_g_o(F.relu(self.w_s_o(o_a_view) + self.w_s_o_(e_obj))), dim=-2)

Am I right? Look forward to your valuable apply! Thanks a lot!

**Comment by tuyunbin** [comment]
Yes, I have same confusion and I have a little trouble to align the code with paper equation.

**Comment by tuyunbin** [comment]
@runzeer Have you understood this reasoning module? I try my best to read it, but I cannot align these codes with the Eq.(3) ~ Eq.(9).
